### PR TITLE
Consolidate nn.module use into simple and GMBlock

### PR DESCRIPTION
**Pull request by csubich**
This pull request:

* Expands logging to log gradient magnitude and wallclock training step time
* Modifies the progress bars:
  * The training progress bar is based on the total number of training steps, not per-epoch
  * The validation progress bar is persistent in Jupyter notebooks
  * The useless 'v_num' tag is removed, freeing up space
* Adjusts type annotations for consistency
* Builds convenience nn.Modules under `simple_blocks.py` for all current nearly-linear modules and a few new ones (like learned global bias)
* Creates the generic multilayer block GMBlock, which is capable of replacing all current CLP (and other nn.Module) calls in the non-variational form of paradis

### CODE REVIEW

**Review by carlospereiraf — APPROVED** [review]
Thanks Christopher, also thanks+ for adding missing typing in many methods
